app/models/campaign.rb

```diff
@@ -0,0 +1,63 @@
+class Campaign < ApplicationRecord
+  include AASM
+
+  has_many :campaign_templates
+  has_many :templates, through: :campaign_templates
+  has_many :campaign_contacts
+  has_many :contacts, through: :campaign_contacts
+
+  validates :name, presence: { message: 'Campaign name is required' }
+  validates :campaign_type, presence: { message: 'Campaign type is required' }
+  validates :email_limit, presence: { message: 'Email limit is required' }
+  validates :start_time, presence: { message: 'Start time is required' }
+  validates :end_time, presence: { message: 'End time is required' }
+  validates :campaign_run_time, presence: { message: 'Campaign run time is required' }
+  validates :batch_contact, presence: { message: 'Campaign batch contact is required' }
+
+  after_commit :schedule_email_job
```

Comment:
Are you saying that whenever I will pause the campaign and again move it to saved, then it will sending the emails from the start ?